bin/cli.ts

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env node
+
+import cac from 'cac';
+import { script, Completion } from '../src/index.js';
+import tab from '../src/cac.js';
+
+import { setupCompletionForPackageManager } from './completion-handlers';
+
+const packageManagers = ['npm', 'pnpm', 'yarn', 'bun'];
+const shells = ['zsh', 'bash', 'fish', 'powershell'];
+
+const cli = cac('tab');
+
+cli
+  .command(
+    '<packageManager> complete',
+    'Process completion requests from shell'
+  )
+  .allowUnknownOptions()
+  .action(async (packageManager) => {
+    if (!packageManagers.includes(packageManager)) {
+      console.error(`Error: Unsupported package manager "${packageManager}"`);
+      console.error(
+        `Supported package managers: ${packageManagers.join(', ')}`
+      );
+      process.exit(1);
+    }
+
+    const dashIndex = process.argv.indexOf('--');
+    if (dashIndex !== -1) {
+      const completion = new Completion();
+      setupCompletionForPackageManager(packageManager, completion);
+      const toComplete = process.argv.slice(dashIndex + 1);
+      await completion.parse(toComplete);
+      process.exit(0);
+    } else {
+      console.error(`Error: Expected '--' followed by command to complete`);
+      console.error(
+        `Example: npx @bombsh/tab ${packageManager} complete -- command-to-complete`
+      );
+      process.exit(1);
+    }
+  });
+
+cli
+  .command(
+    '<packageManager> <shell>',
+    'Generate shell completion script for a package manager'
+  )
+  .action(async (packageManager, shell) => {
+    if (shell === 'complete') {
```

Comment:
hmm why do we have complete in `npx @bombsh/tab pnpm zsh complete`? that's invalid.